### PR TITLE
kcp apiExport and resourceSchema

### PR DESCRIPTION
**Pull request by joejstuart**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -12,6 +12,9 @@ else
 GOBIN=$(shell go env GOBIN)
 endif
 
+# CONTROL_PLANE defines the type of cluster that will be used. Possible values are kubernetes (default) and kcp.
+CONTROL_PLANE ?= kubernetes
```
Comment (zregvart) [inline Makefile:16]: Having kcp off by default makes it easier to overlook when updating the CRD, I'd have it on by default & mandatory.

**Review by zregvart — APPROVED** [review]
I'd make generating kcp manifests mandatory.

There is a bunch of stuff that we can improve in the Makefile, e.g. installing the required kubectl kcp plugin, some of the bash could be Makefile targets (but that's just bike shedding).

Let's get this merged and followup if we feel like it...

**Comment by joejstuart** [comment]
@zregvart I bumped the go version and am installing kubectl-kcp if you want to re-review.

File: .github/workflows/checks.yaml [hunk 2]
```diff
@@ -19,7 +19,10 @@ jobs:
       - name: Setup Go environment
         uses: actions/setup-go@v3.0.0
         with:
-          go-version: 1.17
+          go-version: 1.18
+
+      - name: install kubectl-kcp
+        run: curl -L https://github.com/kcp-dev/kcp/releases/download/v0.8.2/kubectl-kcp-plugin_0.8.2_linux_amd64.tar.gz | tar -xz -C $HOME/go
```
Comment (lcarva) [inline .github/workflows/checks.yaml:25]: Any particular reason `$HOME/go` is used? Maybe `/usr/local/bin` is a better place?

**Review by lcarva — APPROVED** [review]

File: .github/workflows/checks.yaml (hunk 2 above)
Comment (joejstuart) [reply .github/workflows/checks.yaml:25]: I thought plugins had to go there. I updated it to `/usr/local/bin`

Comment (lcarva) [reply .github/workflows/checks.yaml:25]: Just somewhere in `$PATH`, I think.

**Review by zregvart — APPROVED** [review]

**Comment by zregvart** [comment]
I don't think we need to wait for Pipelines as Code CI to finish...

**Comment by zregvart** [comment]
> I don't think we need to wait for Pipelines as Code CI to finish...

Yeah, IIRC there was an issue updating the GitHub Check status, I see the pipeline finished successfully...

**Pull request merged**